packages/transaction-controller/src/TransactionController.ts

```diff
@@ -644,21 +674,21 @@ export class TransactionController extends BaseController<
 
       const chainId = parseInt(currentChainId, undefined);
       const { approved: status } = TransactionStatus;
-
-      const txNonce =
-        nonce ||
-        (await query(this.ethQuery, 'getTransactionCount', [from, 'pending']));
+      let nonceToUse = nonce;
+      // if a nonce already exists on the transactionMeta it means this is a speedup or cancel transaction
+      // so we want to reuse that nonce and hope that it beats the previous attempt to chain. Otherwise use a new locked nonce
+      if (!nonceToUse) {
+        nonceLock = await this.nonceTracker.getNonceLock(from);
+        nonceToUse = nonceLock.nextNonce.toString(16);
+      }
 
       transactionMeta.status = status;
-      transactionMeta.transaction.nonce = txNonce;
+      transactionMeta.transaction.nonce = addHexPrefix(nonceToUse);
       transactionMeta.transaction.chainId = chainId;
 
       const baseTxParams = {
         ...transactionMeta.transaction,
         gasLimit: transactionMeta.transaction.gas,
-        chainId,
-        nonce: txNonce,
-        status,
```

Comment:
Was the removal of `chainId` and `status` intentional? I don't see an obvious connection to the other changes

Reply:
well the `chainId` is actually just added on line 690 since it is being spread from the `...transactionMeta.transaction` object where it is added on line 687. But the status is removed now... which was not intentional. Not sure if it should be on the `baseTxParams`. Let me take another look.

Reply:
Actually it doesn't seem the either `chainId` or `status` should be on the `txParams` object. Both should live on the meta object?

Reply:
Ah I see, I missed that `chainId` was already set. So that line was redundant.

For `status` I'd recommend leaving it for now, to keep this PR more focused. We can consider removing that in a separate PR. I agree that it does seem strange though.

Reply:
Doesn't really matter if they are there but they're not doing anything as far as I can tell? TxParams are passed to `prepareUnsignedEthTx` where they are passed to `TransactionFactory.fromTxData` which expects an object like:
```
export declare type TxData = {
    /**
     * The transaction's nonce.
     */
    nonce?: BNLike;
    /**
     * The transaction's gas price.
     */
    gasPrice?: BNLike;
    /**
     * The transaction's gas limit.
     */
    gasLimit?: BNLike;
    /**
     * The transaction's the address is sent to.
     */
    to?: AddressLike;
    /**
     * The amount of Ether sent.
     */
    value?: BNLike;
    /**
     * This will contain the data of the message or the init of a contract.
     */
    data?: BufferLike;
    /**
     * EC recovery ID.
     */
    v?: BNLike;
    /**
     * EC signature parameter.
     */
    r?: BNLike;
    /**
     * EC signature parameter.
     */
    s?: BNLike;
    /**
     * The transaction type
     */
    type?: BNLike;
};
```

Reply:
Hmm, yeah I guess you're right. I was concerned it'd affect other parts of mobile if these properties were set in the `txParams` part of the transaction in state, but this variable doesn't seem to be tied to state either, or even any parameter being passed in.

Reply:
Shall I just remove them then?

Reply:
I didn't finish reading down into the `@ethereumjs/tx` library, but it seemed like they weren't used anywhere. As long as we're confident this object isn't exposed at all and that the property isn't used, it seems fine to remove it.

Reply:
actually good thing I checked it looks like `chainId` is used in there even though its not typed to expect it...

Reply:
here for instance:  https://github.com/ethereumjs/ethereumjs-monorepo/blob/60995fb50aa56e5749d471eb7fa424403c3e6249/packages/tx/src/eip2930Transaction.ts#L138

Reply:
Yeah, that one does seem to be needed. But since we're already adding it on the earlier line, this should still be OK